### ISSUE TITLE
Rails 7.0.3 Db migration with multiple Databases disturbing tasks name change when using database_tasks: in database.yml

### ISSUE THREAD

**Issue by net1957** [rails/rails#45552]
### Steps to reproduce
with the following databse.yml
```ruby
default: &default
  adapter: sqlite3
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
  timeout: 5000

development:
  primary:
    <<: *default
    database: db/development.sqlite3
  comdev_fr:
    <<: *default
    migrations_paths: db/migrate_comdev_fr
    database: db/development_comdev_fr.sqlite3
```
the bin/rails -T return (filterer on db:migrate):

rails db:migrate                         # Migrate the database (options: V...
rails db:migrate:comdev_fr               # Migrate comdev_fr database for c...
rails db:migrate:down                    # Runs the "down" for a given migr...
rails db:migrate:primary                 # Migrate primary database for cur...
rails db:migrate:redo                    # Rolls back the database one migr...
rails db:migrate:redo:comdev_fr          # Rolls back comdev_fr database on...
rails db:migrate:redo:primary            # Rolls back primary database one ...
rails db:migrate:status                  # Display status of migrations
rails db:migrate:status:comdev_fr        # Display status of migrations for...
rails db:migrate:status:primary          # Display status of migrations for...
rails db:migrate:up                      # Runs the "up" for a given migrat...

in most update/install script we use only the tasks related to the primary db:
rails db:migrate:primary                 # example

With the following databse.yml
```ruby
default: &default
  adapter: sqlite3
  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
  timeout: 5000

development:
  primary:
    <<: *default
    database: db/development.sqlite3
  comdev_fr:
    <<: *default
    database_tasks: false
    migrations_paths: db/migrate_comdev_fr
    database: db/development_comdev_fr.sqlite3
```
the bin/rails -T return (filterer on db:migrate):

rails db:migrate                         # Migrate the database (options: V...
rails db:migrate:down                    # Runs the "down" for a given migr...
rails db:migrate:redo                    # Rolls back the database one migr...
rails db:migrate:status                  # Display status of migrations
rails db:migrate:up                      # Runs the "up" for a given migrat...

and 
rails db:migrate:primary command is not found and so our script fail.

I have tried with more DB and as soon only 1DB has database_tasks: true (the default),
the tasks related to multi-db don't exist any more

I would be nice that in this case
the task related to the primary db don't disappear.

rails db:migrate:primary  # should be here

*** Why?
In the future if we add a new DB that support migrations, the commands for migrating the primary DB
change from 
rails db:migrate
to
rails db:migrate:primary

and most time migration are done DB by DB in separate scripts, that we need to change.

So it would be nice  to allow the  task rails db:migrate:xxxx to exist even if it's the only DB with database_tasks: true (the default)

### System configuration
**Rails version**: 7.0.3

**Ruby version**: 3.0.4

**Comment by eileencodes** [comment]
Hey @net1957 this is an interesting use case. If there's only one database (in a single db app) we never show the individual tasks for primary because that would not be useful. While I think I could fix this I'm not sure that it's correct to.

If you have only one database with tasks turned on and the others have tasks turned off there is no functional difference between `db:migrate` and `db:migrate:primary`. By showing both tasks we're indicating there is a functional difference which could be confusing for users. 

If I fix this I guarantee we're going to get bug reports with ":primary tasks showing when there's only one database with tasks turned on".  I get that this breaks your script, but I'm also not sure of a use case for wanting to run migrate on one db and not another. These were designed mostly for running tasks in development. It's not often useful for an application to run pending migrations on only one db and not on another (because the app will error when queries on those tables with pending migrations are run).

Can you help me understand the use case for needing the named tasks in a script?

**Comment by net1957** [comment]
The problem appear after adding database_tasks: false to all databases not fully managed by Rails.
This was added to prevent any Rails modifications.
Before this change the task to migrate the primary DB was: 
  rails db:migrate:primary
After the change the task to migrate the primary DB is :
  rails db:migrate

The script that run the migration (managed by  the DB teams) failed and we did have a long time to understand.

This application is often changed to add or remove DB and it happens sometimes that we add a new table managed by rails to this new DB. In this case the migration task name change for the first DB ans this disturb us.

I can understand it's a corner case but perhaps AR should have a parameter to generate the xx:db in all case, off by default.

Regards

**Comment by eileencodes** [comment]
> The problem appear after adding database_tasks: false to all databases not fully managed by Rails

Yes but you're essentially telling Rails there's only one database so we never create the task for `db:migrate:primary`.

Basically the problem is that we can't know at this point in the Rails code whether you have a single db app or you have 2 databases and one of them has `database_tasks` set to false.

I considered just hiding the tasks from `rake -T` instead of skipping generating them at all but that would have the risk that someone types a destructive command and ends up confused why it ran the command.

> I can understand it's a corner case but perhaps AR should have a parameter to generate the xx:db in all case, off by default.

I think there's a PR to add an environment variable for the database name instead of the named tasks. It ended up stale because of conflicts and my availability at the time but it's still on my radar to merge when I get around to it. I think this would solve your case. Let me know what you think.

**Comment by schmijos** [comment]
The same issue occurred to me just now after a Rails 7.0 upgrade. We control migrations for one database but not for the other. Our Capistrano deployment script failed:

```
set :migration_command, "db:migrate:primary" # we don't want the Odoo DB to be touched in any way
```

So I had to get rid of the double-security-net.
